huaweicloud/services/modelarts/resource_huaweicloud_modelarts_notebook_mount_storage.go

```diff
@@ -0,0 +1,247 @@
+		return fmtp.DiagErrorf("The notebook id=%s must be running, now is =%s", notebookId, notebookDetail.Status)
+	}
+
+	// mount storage
+	opts := notebook.MountStorageOpts{
+		Category:  d.Get("type").(string),
+		MountPath: d.Get("mount_path").(string),
+		Uri:       d.Get("uri").(string),
+	}
+
+	rs, err := notebook.Mount(client, notebookId, opts)
+	if err != nil {
+		return fmtp.DiagErrorf("Error mounting storage to ModelArts notebook id=%s: %s", notebookId, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", notebookId, rs.Id))
+	d.Set("mount_id", rs.Id)
+
+	err = waitingNotebookForMount(ctx, client, notebookId, rs.Id, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceNotebookMountStorageRead(ctx, d, meta)
+}
+
+func resourceNotebookMountStorageRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ModelArtsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ModelArts v1 client, err=%s", err)
+	}
+
+	notebookId, mountId, err := ParseMountFromId(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	detail, err := notebook.GetMount(client, notebookId, mountId)
+	if err != nil {
+		return common.CheckDeletedDiag(d, parseModlArtsErrorToError404(err), "ModelArts notebook's mount storage")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("notebook_id", notebookId),
+		d.Set("mount_id", mountId),
+		d.Set("uri", detail.Uri),
+		d.Set("mount_path", detail.MountPath),
+		d.Set("type", detail.Category),
+		d.Set("status", detail.Status),
+	)
+
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.DiagErrorf("Error setting ModelArts notebook's mount storage fields: %s", mErr)
+	}
+
+	return nil
+}
+
+func resourceNotebookMountStorageDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ModelArtsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ModelArts v1 client, err=%s", err)
+	}
+
+	notebookId, mountId, err := ParseMountFromId(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = notebook.DeleteMount(client, notebookId, mountId)
+	if err != nil {
+		return fmtp.DiagErrorf("delete ModelArts notebook's mount storage failed. %q:%s", d.Id(), err)
+	}
+
+	err = waitingNotebookMountForDeleted(ctx, client, notebookId, mountId, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId("")
+
+	return nil
+}
+
+func waitingNotebookForMount(ctx context.Context, client *golangsdk.ServiceClient, notebookId, mountId string,
+	timeout time.Duration) error {
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{"MOUNTING"},
+		Target:  []string{"MOUNTED"},
+		Refresh: func() (interface{}, string, error) {
```

Comment:
I think you can merge the refresh function between waitingNotebookForMount and waitingNotebookMountForDeleted

Reply:
pend, target and refresh  is different，too few in common